src/anim/anim.js

```diff
@@ -717,6 +717,34 @@ Object.assign(pc, function () {
                     }
                 };
                 return new pc.AnimTarget(func, 'vector', morphInstance.morph._targets.length);
+            },
+            'materialTexture': function (node, textureName) {
+                var object = node;
+                while (object && object.constructor !== pc.Entity) {
+                    object = object.parent;
+                }
+                if (!object ||
+                    !object.model ||
+                    !object.model.model ||
+                    !object.model.model.morphInstances) {
+                    return null;
+                }
+                var meshInstances = object.model.meshInstances;
+                var meshInstance;
+                for (var i = 0; i < meshInstances.length; ++i) {
+                    if (meshInstances[i].node.name === node.name) {
```

Comment:
is this name compare needed? We take meshInstance from the node, so I assume they all point to the same node?

Reply:
We get the mesh instance from the nodes ancestor model. But that model may contain multiple mesh instances so this finds the one we're looking for.